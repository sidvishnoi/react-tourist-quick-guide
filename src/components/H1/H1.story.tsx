import { color, text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import * as React from 'react';
import H1 from '.';

const stories = storiesOf('Atoms/H1', module);
stories.addDecorator(withKnobs);

stories.add('simple', () => {
  return (
    <H1
      color={color('color', '#000')}
      borderColor={color('border-color', '#000')}
    >
      {text('text', 'Hello World')}
    </H1>
  );
});
