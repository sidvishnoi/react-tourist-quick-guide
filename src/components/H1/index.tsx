import styled from 'styled-components';

export interface H1Props {
  color?: string;
  borderColor?: string;
}

const H1 = styled.h1<H1Props>`
  --border-color: ${props => props.borderColor || '#000'};
  color: ${props => props.color || '#000'};
  padding: 0.5em;
  text-align: center;
  border-top: 2px solid var(--border-color);
  border-bottom: 2px solid var(--border-color);
  max-width: 95%;
  margin: 0.2em auto;
`;

export default H1;
