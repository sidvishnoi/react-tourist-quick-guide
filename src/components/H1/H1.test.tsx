import 'jest-styled-components';
import React = require('react');
import * as renderer from 'react-test-renderer';

import H1 from '.';

describe('Component - H1', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<H1>PASS</H1>);
    expect(tree).toMatchSnapshot();
  });
});
