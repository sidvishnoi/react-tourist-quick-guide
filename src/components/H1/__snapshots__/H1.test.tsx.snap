// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - H1 matches snapshot 1`] = `
.c0 {
  --border-color: #000;
  color: #000;
  padding: 0.5em;
  text-align: center;
  border-top: 2px solid var(--border-color);
  border-bottom: 2px solid var(--border-color);
  max-width: 95%;
  margin: 0.2em auto;
}

<h1
  className="c0"
>
  PASS
</h1>
`;
